### DISCUSSION TITLE
I want to store scrollbar instance

### DISCUSSION THREAD

**Discussion by blitzve0** [idiotWu/smooth-scrollbar#447] [Q&A] [answered]
**Summary**
Hi, I am using this on _ReactJS_ and I need to store scrollbar instance for use outside _useEffect_ function.

**Problem**
I want to reset the scroll position when I change/Switch Page. Currently If if change page then scroll position is not changing

**Current Solution**
I am storing instance in _useState_ Hook but it can be very dangerous as it will cause _huge performance_ issues if there is too much data in the instance and I have used this Component on more than 40+ Pages
![Screenshot from 2022-02-15 02-55-47](https://user-images.githubusercontent.com/67477131/153949234-e59cd823-e85a-48a4-be1b-f663a9d141cf.png)

**HELP WANTED**
Option 1 : What is the optimal way to store this instance without causing _Performance Degrading_
Option 2 : Another Trick to reset Scroll Position on Page Switch
Any Solution is Appreciated

**Comment by dolphin-wood** [comment, marked as answer]
Not familiar with react, but I don't think keeping a reference to the scrollbar instance will consume too much memory because the instance always exists unless you destroy the scrollbar via `.destroy()` method.

Regarding your code, as the scrollbar is constructed when `router` changes, I think we can simplify it as:

```js
useLayoutEffect(() => {
  const scrollbar = Scrollbar.init(...);

  scrollbar.setPosition(0, 0); // you can use `.setPosition(0, 0)` instead of `.scrollTo(0, 0, 0)` to reset position

  return () => scrollbar.destroy(); // destroy it if necessary
}, [router]);
```

**Reply by blitzve0** [reply]
Oh Ok I Understood , When needed , Instead of storing a copy of it, i will just set a reference to it with useRef. This setup will work fine for resetting position. Thanks Again 🎊🥳

**Reply by dolphin-wood** [reply]
IMO `useState()` will just work as expected – only the _reference_ to the instance is saved when calling `setScrollbar(scrollbarInstance)`.

**Reply by blitzve0** [reply]
The above solution worked, now I don't need to store it 🥳
